drivers/i2s/i2s_mcux_flexcomm.c

```diff
@@ -0,0 +1,908 @@
+		/* Clear RX error interrupt flag */
+		base->FIFOSTAT = I2S_FIFOSTAT_RXERR(1U);
+		strm = &dev_data->dma_rx;
+		strm->state = I2S_STATE_ERROR;
+	}
+}
+
+static int i2s_mcux_init(const struct device *dev)
+{
+	const struct i2s_mcux_config *cfg = dev->config;
+	struct i2s_mcux_data *const data = dev->data;
+
+	data->dev_clock = device_get_binding(cfg->clock_name);
+	if (data->dev_clock == NULL) {
+		return -ENODEV;
+	}
+
+	cfg->irq_config(dev);
+
+	/* Initialize the buffer queues */
+	k_msgq_init(&data->dma_tx.in_queue, (char *)data->dma_tx.in_msgs,
+		    sizeof(void *), CONFIG_I2S_MCUX_FLEXCOMM_TX_BLOCK_COUNT);
+	k_msgq_init(&data->dma_rx.in_queue, (char *)data->dma_rx.in_msgs,
+		    sizeof(void *), CONFIG_I2S_MCUX_FLEXCOMM_RX_BLOCK_COUNT);
+	k_msgq_init(&data->dma_tx.out_queue, (char *)data->dma_tx.out_msgs,
+		    sizeof(void *), CONFIG_I2S_MCUX_FLEXCOMM_TX_BLOCK_COUNT);
+	k_msgq_init(&data->dma_rx.out_queue, (char *)data->dma_rx.out_msgs,
+		    sizeof(void *), CONFIG_I2S_MCUX_FLEXCOMM_RX_BLOCK_COUNT);
+
+	/* Get the binding to the DMA device */
+	if (data->dma_tx.dma_name != NULL) {
+		/* Get the binding to the DMA device */
+		data->dma_tx.dev_dma = device_get_binding(data->dma_tx.dma_name);
+		if (!data->dma_tx.dev_dma) {
+			LOG_ERR("%s device not found", data->dma_tx.dma_name);
+			return -ENODEV;
+		}
+	}
+
+	if (data->dma_rx.dma_name != NULL) {
+		data->dma_rx.dev_dma = device_get_binding(data->dma_rx.dma_name);
+		if (!data->dma_rx.dev_dma) {
+			LOG_ERR("%s device not found", data->dma_rx.dma_name);
+			return -ENODEV;
+		}
+	}
+
+	data->dma_tx.state = I2S_STATE_NOT_READY;
+	data->dma_rx.state = I2S_STATE_NOT_READY;
+
+	LOG_DBG("%s inited", dev->name);
+
+	return 0;
+}
+
+#define I2S_DMA_CHANNELS(id)				\
+	.dma_tx = {						\
+		.dma_name = UTIL_AND(		\
+			DT_DMAS_HAS_NAME(DT_NODELABEL(i2s##id), tx),	\
+			DT_INST_DMAS_LABEL_BY_NAME(id, tx)),		\
+		.channel = UTIL_AND(		\
+			DT_DMAS_HAS_NAME(DT_NODELABEL(i2s##id), tx),	\
+			DT_INST_DMAS_CELL_BY_NAME(id, tx, channel)),	\
+		.dma_cfg = {					\
+			.channel_direction = MEMORY_TO_PERIPHERAL,	\
+			.dma_callback = i2s_mcux_dma_tx_callback,	\
+			.source_data_size = 1,				\
+			.block_count = 1,		\
+		}							\
+	},								\
+	.dma_rx = {						\
+		.dma_name = UTIL_AND(		\
+			DT_DMAS_HAS_NAME(DT_NODELABEL(i2s##id), rx),	\
+			DT_INST_DMAS_LABEL_BY_NAME(id, rx)),		\
+		.channel = UTIL_AND(		\
+			DT_DMAS_HAS_NAME(DT_NODELABEL(i2s##id), rx),	\
+			DT_INST_DMAS_CELL_BY_NAME(id, rx, channel)),	\
+		.dma_cfg = {				\
+			.channel_direction = PERIPHERAL_TO_MEMORY,	\
+			.dma_callback = i2s_mcux_dma_rx_callback,	\
+			.source_data_size = 1,				\
+			.block_count = 1,		\
+		}							\
+	}
+
+#define I2S_MCUX_FLEXCOMM_DEVICE(id)					\
+	static void i2s_mcux_config_func_##id(const struct device *dev); \
+	static const struct i2s_mcux_config i2s_mcux_config_##id = {	\
+		.base =							\
+		(I2S_Type *)DT_INST_REG_ADDR(id),			\
+		.clock_name = DT_INST_CLOCKS_LABEL(id),		\
```

Comment:
Trying to phase out DT_INST_CLOCKS_LABEL, see recent changes to other NXP drivers to use `DEVICE_DT_GET`

Reply:
Thank you, I will make this change.

Reply:
Thanks for fixing this, but it was done in the wrong commit. Can you move the change into the commit `drivers: i2s: Add support for I2S driver for LPC devices`

Reply:
Thanks for catching this, I have fixed this.